server/base/src/main/java/org/apache/accumulo/server/util/ratelimit/SharedRateLimiterFactory.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.server.util.ratelimit;
+
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.WeakHashMap;
+import java.util.concurrent.Callable;
+import java.util.concurrent.atomic.LongAdder;
```

Comment:
I was unaware of LongAdder... its really neat.  Yet another nice thing in JDK8 we can't use yet.